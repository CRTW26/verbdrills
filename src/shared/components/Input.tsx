import React from 'react'

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  onChange: (ev: React.ChangeEvent<HTMLInputElement>) => void
}

const Input: React.FC<Props> = ({ onChange, ...rest }) => {
  return (
    <>
      <input onChange={onChange} {...rest} />

      <style jsx>{`
        input {
          min-width: 200px;
          padding: 0.5rem 1rem;
        }
      `}</style>
    </>
  )
}

export default Input
