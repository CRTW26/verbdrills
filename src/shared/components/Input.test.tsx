import { render } from '@testing-library/react'
import Input from './Input'

describe('<Input />', () => {
  test('should render', () => {
    const { asFragment } = render(<Input onChange={jest.fn()} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
