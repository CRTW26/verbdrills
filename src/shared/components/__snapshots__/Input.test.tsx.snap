// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render 1`] = `
<DocumentFragment>
  <input
    class="jsx-1649105467 "
  />
</DocumentFragment>
`;
